From 44c8ca79d4a024e0e83f763c14fdc22a1f955e7b Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Tue, 23 Feb 2021 14:14:52 +0200
Subject: [PATCH] [SW-20154] gaudi2_arc_gnu: asic: fix compilation error

A variable "factor" has a missing declaration and hence
throws a compilation error. This patch fixes it

Change-Id: I7a27c6e131c6be2c8be880e1a8923f53255bbe73
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/gaudi2_arc_gnu/asic.c b/gaudi2_arc_gnu/asic.c
index c7c916d..818e386 100644
--- a/gaudi2_arc_gnu/asic.c
+++ b/gaudi2_arc_gnu/asic.c
@@ -65,7 +65,7 @@
 	u32 spi_clk_freq = hl_pll_get_freq(mmPSOC_CPU_PLL_CTRL_BASE,
 					   HL_PLL_DIV3);
 
-	factor = spi_clk_freq / HL_SPI_FLASH_MAX_FREQ;
+	u32 factor = spi_clk_freq / HL_SPI_FLASH_MAX_FREQ;
 
 	return factor > 2 ? factor : 2;
 }
